src/libs/actions/IOU.js

```diff
@@ -722,11 +723,326 @@ function payIOUReport({
+                    errors: {
+                        [DateUtils.getMicroseconds()]: Localize.translateLocal('iou.error.genericCreateFailureMessage'),
+                    },
+                },
+            },
+        },
+    ];
+
+    // Now, let's add the data we need just when we are creating a new chat report
+    if (isNewChat) {
+        const optimisticCreateAction = ReportUtils.buildOptimisticCreatedReportAction(recipientEmail);
+
+        // Change the method to set for new reports because it doesn't exist yet, is faster,
+        // and we need the data to be available when we navigate to the chat page
+        optimisticData[0].onyxMethod = CONST.ONYX.METHOD.SET;
+        optimisticData[0].value = {
+            ...optimisticData[0].value,
+            pendingFields: {createChat: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD},
+        };
+        optimisticData[1].onyxMethod = CONST.ONYX.METHOD.SET;
+        optimisticData[1].value = {
+            ...optimisticCreateAction,
+            ...optimisticData[1].value,
+        };
+        optimisticData[2].onyxMethod = CONST.ONYX.METHOD.SET;
+    }
+
+    return {
+        params: {
+            iouReportID: optimisticIOUReport.reportID,
+            chatReportID: chatReport.reportID,
+            reportActionID: optimsticReportActionID.reportActionID,
+            paymentMethodType,
+            transactionID: optimsticReportActionID.originalMessage.IOUTransactionID,
+            clientID: optimsticReportActionID.sequenceNumber,
+            newIOUReportDetails,
+        },
+        optimisticData,
+        successData,
+        failureData,
+    };
+}
+
+/**
+ * @param {Object} chatReport
+ * @param {Object} iouReport
+ * @param {Object} recipient
+ * @param {String} paymentMethodType
+ * @returns {Object}
+ */
+function getPayMoneyRequestParams(chatReport, iouReport, recipient, paymentMethodType) {
+    const newSequenceNumber = Report.getMaxSequenceNumber(chatReport.reportID) + 1;
+
+    const optimsticReportActionID = ReportUtils.buildOptimisticIOUReportAction(
+        newSequenceNumber,
+        CONST.IOU.REPORT_ACTION_TYPE.PAY,
+        iouReport.total,
+        iouReport.currency,
+        '',
+        [recipient],
+        paymentMethodType,
+        '',
+        iouReport.reportID,
+    );
+
+    // First, add data that will be used in all cases
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${chatReport.reportID}`,
+            value: {
+                ...chatReport,
+                lastVisitedTimestamp: Date.now(),
+                lastReadSequenceNumber: newSequenceNumber,
+                maxSequenceNumber: newSequenceNumber,
+                lastMessageText: optimsticReportActionID.message[0].text,
+                lastMessageHtml: optimsticReportActionID.message[0].html,
+                hasOutstandingIOU: false,
+                iouReportID: iouReport.reportID,
```

Comment:
Lets also maybe merge this PR into this branch https://github.com/Expensify/App/pull/12759/files, we are waiting there for Web E deploy to production, but it could be handy here too

Reply:
This is the only change left, what's the decision here?

Reply:
@mountiny I can see that PR just got merged, but I'm not sure I need that logic in SendMoney and payMoneyRequest.

For SendMoney, we are always creating an IOUReport and then settling up. For payMoneyRequest, we always know we have an IOUReport to settle. What do you think?

@luacmartins clearing the iouReportID makes sense to me as well.

Reply:
> but I'm not sure I need that logic in SendMoney and payMoneyRequest.

I agree that that logic doesn't seem necessary here.

>clearing the iouReportID makes sense to me as well.

Yea, let's clear the `iouReportID`